### PR TITLE
[arithmetic-side-effects] More non-overflowing ops

### PR DESCRIPTION
**Pull request by c410-f3r**
* Adding or Subtracting 0
* Division and Module of anything other than 0
* Multiplying 1 or 0

changelog: [arithmetic-side-effects] More non-overflowing operations

### CODE REVIEW

File: clippy_lints/src/operators/arithmetic_side_effects.rs [hunk 1]
```diff
@@ -48,9 +48,21 @@ impl ArithmeticSideEffects {
...
+        }
+        if let hir::BinOpKind::Div | hir::BinOpKind::Rem = op.node
+            && let hir::ExprKind::Lit(ref lit) = rhs.kind
+            && !matches!(lit.node, ast::LitKind::Int(0, _))
+        {
+            return true;
+        }
+        if let hir::BinOpKind::Mul = op.node
+            && let hir::ExprKind::Lit(ref lit) = rhs.kind
+            && let ast::LitKind::Int(0, _) | ast::LitKind::Int(1, _) = lit.node
```
Comment (llogiq) [inline clippy_lints/src/operators/arithmetic_side_effects.rs:65]:
```suggestion
            && let ast::LitKind::Int(0 | 1, _) = lit.node
```

**Review by llogiq — COMMENTED** [review]
This looks good. Thank you! I just have a very small suggestion, otherwise r=me.

**Comment by c410-f3r** [comment]
Done

Thank you @llogiq

**Comment by llogiq** [comment]
Thank you!

@bors r+

**Pull request merged**